frappe/custom/doctype/custom_field/test_custom_field.py

```diff
@@ -37,3 +36,26 @@ def test_create_custom_fields(self):
 		self.assertTrue(frappe.db.exists("Custom Field", "Address-_test_custom_field_1"))
 		self.assertTrue(frappe.db.exists("Custom Field", "Address-_test_custom_field_2"))
 		self.assertTrue(frappe.db.exists("Custom Field", "Contact-_test_custom_field_2"))
+
+	def test_custom_field_sorting(self):
+		custom_fields = {
+			"ToDo": [
+				{"fieldname": "a_test_field", "insert_after": "b_test_field"},
+				{"fieldname": "b_test_field", "insert_after": "status"},
+				{"fieldname": "c_test_field", "insert_after": "unknown_custom_field"},
+				{"fieldname": "d_test_field", "insert_after": "status"},
```

Comment:
You'll have to clean up these... postgres tests failing because of them. 


Example: https://github.com/frappe/frappe/blob/e564dcb3fcea09ea5c471c7f457cadcb93e64285/frappe/automation/doctype/assignment_rule/test_assignment_rule.py#L277